oshi-core/src/main/java/oshi/software/common/AbstractOperatingSystem.java

```diff
@@ -103,7 +103,7 @@ private int queryPlatformBitness() {
 
     @Override
     public List<OSProcess> getProcesses(Predicate<OSProcess> filter, Comparator<OSProcess> sort, int limit) {
-        return queryAllProcesses().stream().filter(filter == null ? ALL_PROCESSES : filter)
+        return queryAllProcesses().stream().parallel().distinct().filter(filter == null ? ALL_PROCESSES : filter)
```

Comment:
put distinct before parallel or you lose the benefits.

But in this case you shouldn't use it as there's a limit at the end and random order (possible with no sorting option) would lead to nondeterministic results.